src/wallet/coinselection.cpp

```diff
@@ -470,6 +471,7 @@ CAmount SelectionResult::GetSelectionWaste(CAmount change_cost, CAmount target,
         waste += change_cost;
     } else {
         // When we are not making change (change_cost == 0), consider the excess we are throwing away to fees
+        CAmount selected_effective_value = use_effective_value ? GetSelectedEffectiveValue() : GetSelectedValue();
```

Comment:
It seems to me in the case with the excess `bump_fee_group_discount` will be canceled.

first you subtract the discount `waste -= bump_fee_group_discount;` but then `GetSelectedEffectiveValue()` adds it back `bump_fee_group_discount`

I think you only need to do `waste -= bump_fee_group_discount;` when there is a change, but I'm not absolutely sure.

Could you add a test for the waste calculation with bump_fee and excess?

Reply:
I’ve added a test. I think the canceling out of the `bump_fee_group_discount` is expected: both the bump fees and the excess count towards waste in full, so whether we internally attribute the additional fees to the group discount or the excess should not matter for the overall waste score.